backend/scoreserver/domain/notice.go

```diff
@@ -0,0 +1,171 @@
+package domain
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/gofrs/uuid/v5"
+)
+
+type Notice notice
```

Comment:
やりたいことは以下じゃないかな
```suggestion
type Notice = notice
```
他の部分(user とか)ではある基本的な型(User)でできることを拡張された型(TeamMember)でも可能にするために埋め込みを使っている。埋め込まれたフィールドもプライベートになってほしいため`type user`で構造体を定義して`type User=user`にしたり，`type User`で定義してその逆の形にしたりしている(ここにブレがあるのはちゃんとしていなくてすみません)。ここで重要なのはパブリックな型とプライベートな型が new type ではなく alias の関係になっていることで，これによって片方で定義されたメソッドをもう片方でも使うことができて埋め込みが機能する